mmv1/third_party/terraform/services/networksecurity/resource_network_security_firewall_endpoint_association_test.go.erb

```diff
@@ -50,66 +111,71 @@ func TestAccNetworkSecurityFirewallEndpointAssociations_basic(t *testing.T) {
 	})
 }
 
-func testAccNetworkSecurityFirewallEndpointAssociation_basic(randomSuffix string, orgId string) string {
+func testAccNetworkSecurityFirewallEndpointAssociation_basic(randomSuffix string, orgId string, billingProjectId string, projectNumberId string) string {
 	return fmt.Sprintf(`
 resource "google_compute_network" "foobar" {
-	provider                = google-beta
-    name                    = "tf-test-my-vpc%s"
+	provider                  = google-beta
+    name                    = "tf-test-my-vpc%[2]s"
     auto_create_subnetworks = false
 }
 
 resource "google_network_security_firewall_endpoint" "foobar" {
-    provider = google-beta
-    name     = "tf-test-my-firewall-endpoint%s"
-    parent   = "organizations/%s"
-    location = "us-central1-a"
+    provider           = google-beta
+    name               = "tf-test-my-firewall-endpoint%[2]s"
+    parent             = "organizations/%[3]s"
+    location           = "us-central1-a"
+    billing_project_id = "%[1]s"
 }
 
 # TODO: add tlsInspectionPolicy once resource is ready
 resource "google_network_security_firewall_endpoint_association" "foobar" {
     provider          = google-beta
-    name              = "tf-test-my-firewall-endpoint%s"
-    parent            = "organizations/%s"
+    name              = "tf-test-my-firewall-endpoint-association%[2]s"
+    parent            = "projects/%[1]s"
     location          = "us-central1-a"
     firewall_endpoint = google_network_security_firewall_endpoint.foobar.id
-    network           = google_compute_network.foobar.id
+    network           = "projects/%[4]s/global/networks/tf-test-my-vpc%[2]s"
```

Comment:
Why doesn't the network id work?

Reply:
There was some issue with using the network id directly in that the provider would generate a name using the project id, while the resource API would return the name using the project number. This caused the tests to fail as It resulted in changes in the plan after applying the test config:

```
vcr_utils.go:152: Step 1/4 error: After applying this test step, the plan was not empty.
        stdout:
        [...]
        ~ network           = "projects/[PROJECT_NUMBER]/global/networks/[NETWORK_NAME]" -> "projects/[PROJECT_ID]/global/networks/[NETWORK NAME]"
        [...]
```

I think this issue might require some further testing.

Reply:
That's a violation of the API design guidelines per https://google.aip.dev/cloud/2510#google-apis, so that's a bug w/ the API if so

Reply:
I am aligning this with a Google PM. I will be back as soon as I get an update.

Reply:
I got an update from a Google PM, they are working on a fix to persist and return user-provided values in resource references. ETA is uncertain but should be around a week.

I reverted my changes to "endpoint" field for now, I'l give a update whenever their changes go into prod and our tests can pass.

Reply:
Thanks!

Reply:
We can mark the field a ResourceRef instead of a string, that will cause us to only compare the shortname and ignore the project value difference.

Reply:
I just got confirmation that the API fixes reached prod, the tests seems to be working now.